src/order/order-request-sender.ts

```diff
@@ -48,13 +49,19 @@ export default class OrderRequestSender {
     submitOrder(body: InternalOrderRequestBody, { headers, timeout }: SubmitOrderRequestOptions = {}): Promise<Response<InternalOrderResponseBody>> {
         const url = '/internalapi/v1/checkout/order';
 
-        return this._requestSender.post(url, {
+        return this._requestSender.post<InternalOrderResponseBody>(url, {
             body,
             headers: omitBy({
                 'X-Checkout-Variant': headers && headers.checkoutVariant,
                 ...SDK_VERSION_HEADERS,
             }, isNil),
             timeout,
+        }).catch(error => {
+            if (error.status === 500 && error.body.type === 'tax_provider_unavailable') {
```

Comment:
Not sure if `error.status === 500` is necessary. Would love to hear any advice🙇‍♂️

Reply:
I think just the `error.body.type` should be good, you can check if the error is a type of `RequestError` if you really want to check the error type.

Reply:
Thanks. Status code check removed.